### DISCUSSION TITLE
Help needed for a WIFI Roaster

### DISCUSSION THREAD

**Discussion by Diomioleo** [artisan-roaster-scope/artisan#714] [Q&A] [answered]
Hi I am using a Santoker Q5 roaster,  I am connecting this roaster through wifi, the raster is working like a wifi host，
after I connect my laptop to this roaster's wifi hotspot.
 
the roast IP address is 10.10.100.1,  and my laptop will be allocated a dynamic IP usually like 10.10.100.*

I would like to ask for some help in this case how can I set up a new device through the  configure-device setting, so that the the machine can work with Artisan.

any help will be appreciated, thanks.

**Comment by MAKOMO** [comment, marked as answer]
The Santoker Q5 is not supported by Artisan. Contact the producer of this device to support the Artisan team in implementing the required structures.

**Reply by Diomioleo** [reply]
thank you Makomo!

**Reply by acx4** [reply]
Hi, I have the Q5 and the Artisan version 2.8.2 is working great.  Even when running the Santoker app at the same time, Artisan will control the buttons on the other Santoker app (charge, dry, first crack...etc...)

**Comment by lins05** [comment]
Hey @MAKOMO, I'm also using a Santoker wifi roaster and I'd like to integrate it with artisan so that I could control the heater/fan etc from within artisan. Are you open to accept a patch for this? 

The implementation would be quite like the part for controlling Aillio Bullet R1 (except the latter uses usb control while here we're going to send tcp packets).

**Reply by MAKOMO** [reply]
Please submit a PR

**Reply by acx4** [reply]
Hi, I have the Q5 and the Artisan version 2.8.2 is working great. Even when running the Santoker app at the same time, Artisan will control the buttons on the other Santoker app (charge, dry, first crack...etc...)

**Reply by Diomioleo** [reply]
how do you setup the SW? thanks.
